clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -1663,8 +1665,8 @@ void FunctionVariableConstraint::handle_params
   // The only case where the params are not equal, and it's not
   // an untype call, is a variadic call. Variadic calls wil not
   // generate deferments.
-  if (paramsEq && To->getDeferredParams().size() == 0
-      && From->getDeferredParams().size() == 0 ) {
+  if (paramsEq || (To->getDeferredParams().size() == 0
+                   && From->getDeferredParams().size() == 0 )) {
```

Comment:
I don't understand this conditional. It's saying: If the parameter counts are equal then we just merge. Fine. 

It's also saying, regardless of whether the param counts are equal, if there are no deferred constraints, then just do merging/transplanting. But this won't work if either `From` or `To` is empty.

I think you want this conditional to be simply `if (paramsEq) {`. (More below)